### ISSUE TITLE
Include tests for checking mass balance calculation

### ISSUE THREAD

**Issue by adehecq** [GlacioHack/xdem#184]
One of the goals of volume.py is to provide all functions needed for the geodetic mass balance calculation.
For now, we don't have a function to do that. I don't know if that's wished or not, as there are many post-processing that could be applied to the DEMs before that, such as filtering, gap-filling etc.
Maybe we could at least provide a function that does the calculation for a region/individual glacier and a gap-free ddem?

Anyway, we should make sure that the functions to calculate the MB are flawless. One of the early issues was that if the uppermost/lowermost elevation band did not contain any valid data, they were not taken into account, which would lead to a wrong mass balance (see PR #118).
Hence, we should have a data set with one (several?) glacier outlines, and a synthetic ddem that varies with altitude and constant within e.g. 50 m elevation bands, so that the exact volume change is known. We could then test that the estimated mass balance is exactly as expected, when one perform a mean/median hypsometric approach.
Any thoughts @erikmannerfelt @rhugonnet ?

**Comment by erikmannerfelt** [comment]
Good idea! I would first prefer to have some kind of synthetic approach which could be parametrized over different nan cases, sizes, and other settings. I know I'm talking a lot about pandas, but I think they provide good examples of a working framework: They don't have a single actual dataset for their entire core functionality!!

Regarding the SMB calculations, they technically exist in `DEMCollection` (although they may also need revision?). I always envisioned that object as a helper, however, not a requirement, so I agree that some functions should be in volume.py as well.

Only partly on topic, with https://github.com/GlacioHack/GeoUtils/pull/185 we can synthesize "glacier" outlines soon! Thanks, @AdrienWehrle!

**Comment by rhugonnet** [comment]
+1 for synthetic approaches!
Fully agree, and we should probably talk in "volume change" terms instead of "mass balance"! This kind of calculation can technically be applied to snow, forest, landslides, etc! :sweat_smile:
